README.md

```diff
@@ -1,36 +1,128 @@
 # `@ubiquity-os/command-ask`
 
-This is a high context aware GitHub organization integrated bot that uses the OpenAI GPT-4o model to provide highly relevant answers to questions and queries in GitHub issues and pull requests.
+The ask command is a high context aware GitHub organization integrated bot that uses OpenRouter.ai to provide highly relevant answers to questions and queries in GitHub issues and pull requests.
 
 ## Usage
 
-In any repository where your Ubiquity OS app is installed, both issues and pull requests alike, you simply mention `@UbiquityOS` with your question or query and using the latest OpenAi GPT-4o model, the bot will provide you with a highly relevant answer.
+In any repository where your UbiquityOS app is installed, both issues and pull requests alike, you simply mention `@UbiquityOS` with your question or query and the bot will provide you with a highly relevant answer using OpenRouter.ai.
 
 ## How it works
 
 With its huge context window, we are able to feed the entire conversational history to the model which we obtain by recursively fetching any referenced issues or pull requests from the chat history. This allows the model to have a very deep understanding of the current scope and provide highly relevant answers.
 
 As it receives everything from discussions to pull request diffs and review comments, it is a highly versatile and capable bot that can assist in a wide range of scenarios.
 
+## Technical Architecture
+
+### System Overview
+
+Command Ask is built as a Cloudflare Worker that integrates with GitHub's webhook system to process issue comments. The system leverages OpenRouter.ai and vector similarity search to provide intelligent responses:
+
+|                   | OpenRouter.ai | Voyage |
+|-------------------|---------------|---------|
+| Embeddings        | ❌            | ✅      |
+| Reranking        | ❌            | ✅      |
+| LLM              | ✅            | ❌      |
+
+```
+┌─────────────────┐         ┌──────────────┐
+│  GitHub Webhook │ ──────> │   Cloudflare │
+│    (Comments)   │         │    Worker    │
+└─────────────────┘         └──────┬───────┘
```

Comment:
```suggestion
┌─────────────────┐         ┌──────────────┐
│  GitHub Webhook │ ──────> │    Github    │
│    (Comments)   │         │    Action    │
└─────────────────┘         └──────┬───────┘
```

Shouldn't this be a GitHub Action, since it's more commonly used than worker deployment in most cases?